src/Backend/Utils.idr

```diff
@@ -75,8 +73,53 @@ flattenMus muName = flattenMu [muName]
   flattenMu names (TProd ts)  = assert_total $ TProd $ map (flattenMu names) ts
   flattenMu names (TMu cs)    = assert_total $ TMu $ map (map (flattenMu ((nameMu cs) :: names))) cs
   flattenMu names (TApp f xs) = assert_total $ TApp f (map (flattenMu names) xs)
+  flattenMu names (TRef n)    = TRef n
 
 -- TODO: is this in a library somewhere?
 mapWithIndexA : Applicative m => {n : Nat} -> (Fin n -> a -> m b) -> Vect n a -> m (Vect n b)
```

Comment:
Is this now unused? Should it replace `traverseWithIndex` in `Backend.Haskell`?

Reply:
Good call, I removed it